php-checks/src/main/java/org/sonar/php/checks/ThisVariableUsageInStaticContextCheck.java

```diff
@@ -49,10 +51,22 @@ public void visitMethodDeclaration(MethodDeclarationTree methodDec) {
 
   @Override
   public void visitVariableIdentifier(VariableIdentifierTree varIdentifier) {
-    if (inStaticContext && "$this".equals(varIdentifier.variableExpression().text())) {
+    if (inStaticContext && !isWithinNonStaticFunctionExpression(varIdentifier) && "$this".equals(varIdentifier.variableExpression().text())) {
```

Comment:
Nitpicking: I suppose that this new condition is slightly more costly than the next one. I would have put the new condition last.